### PR TITLE
optional running either v1 or v2 contract monitoring

### PR DESCRIPTION
**Pull request by swelf19**
What have been done.
Added environment variable BASSET_CONTRACTS_VERSION with default value = "2", witch indicates version of contracts are being monitored.
The correct monitor versions are used based on the value of environment variable.
Updated promtail config to collect only appropriate log entries.
Updated docker-compose, now we have to pass environment variables explicitly, telegram.env is an exception (see readme)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,22 +7,43 @@ TELEGRAM_BOTTOKEN=<telegram_bottoken>
 TELEGRAM_CHAT_ID=<telegram_chat_id>
 ```
 
-Before running the service you have to fill .env file `./docker/env/.lido_terra.env` with the following variables:
+Variables required to service work.\
+You can fill some .env file and pass it to docker-compose service with --env-file argument.
```
Comment (oopcode) [inline README.md:11]: s/You can fill some .env file and pass it/You should pass the .env file/g

Comment (oopcode) [reply README.md:11]: `--env-file` (formatting)

File: collector/monitors/configs_monitor.go [hunk 2]
```diff
@@ -22,37 +22,47 @@ const (
...
-	return []MetricName{
+	providedMetrics := []MetricName{
 		AirDropRegistryConfigCRC32,
-		ValidatorsRegistryConfigCRC32,
-		RewardDispatcherConfigCRC32,
 		HubConfigCRC32,
-		BlunaRewardConfigCRC32}
+		BlunaRewardConfigCRC32,
+	}
+	// ValidatorsRegistry and RewardDispatcher contracts are presented only for v2 contracts,
```
Comment (oopcode) [inline collector/monitors/configs_monitor.go:60]: presented -> present in / relevant for

File: collector/monitors/hub_state_monitor.go [hunk 3]
```diff
@@ -18,17 +18,29 @@ var (
 	BlunaExchangeRate MetricName = "bluna_exchange_rate"
 )
 
-func NewHubStateMonitor(cfg config.CollectorConfig, logger *logrus.Logger) HubStateMonitor {
-	m := HubStateMonitor{
-		State:      &types.HubStateResponseV1{},
+func NewHubStateMonitor(cfg config.CollectorConfig, logger *logrus.Logger) Monitor {
+
+	if cfg.BassetContractsVersion == config.V1Contracts {
```
Comment (oopcode) [inline collector/monitors/hub_state_monitor.go:23]: Please change to a switch with default error ("version unknown"), it's more explicit

File: collector/monitors/validators_repository.go [hunk 4]
```diff
@@ -1,89 +1,16 @@
...
-
-	p := wasm.GetWasmContractsContractAddressStoreParams{}
-	p.SetContext(ctx)
-	p.SetContractAddress(r.hubContract)
-	p.SetQueryMsg(string(reqRaw))
-
-	resp, err := r.apiClient.Wasm.GetWasmContractsContractAddressStore(&p)
-	if err != nil {
-		return nil, fmt.Errorf("failed to process HubWhitelistedValidators request: %w", err)
+func NewValidatorsRepository(cfg config.CollectorConfig) ValidatorsRepository {
```
Comment (oopcode) [inline collector/monitors/validators_repository.go:5]: same thing, a switch with default error would do better

**Review by oopcode — APPROVED** [review]

**Review by pr0n00gler — APPROVED** [review]

**Pull request merged**